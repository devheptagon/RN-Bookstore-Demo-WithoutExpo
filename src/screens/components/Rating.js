import React from 'react'
import PropTypes from 'prop-types'
import StarRating from 'react-native-star-rating'
import { StyleSheet } from 'react-native'
import Theme from '../../constants/Theme'

const Rating = (props) => (
    <StarRating
        disabled={false}
        maxStars={5}
        rating={props.value || 3}
        starSize={13}
        fullStarColor={Theme.ProductList.Item.RatingStarColor}
        containerStyle={styles.ContainerStyle}
    />
)

const styles = StyleSheet.create({
    ContainerStyle: {
        width: '30%',
        marginTop: 5,
        marginBottom: 5
    }
})

Rating.propTypes = {
    value: PropTypes.number
}

export default Rating