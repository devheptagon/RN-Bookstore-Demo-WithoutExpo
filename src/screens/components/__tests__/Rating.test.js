import React from "react";
import Rating from "../Rating";
import renderer from "react-test-renderer";

test("Rating renders correctly", () => {
  const tree = renderer.create(<Rating />).toJSON();
  expect(tree).toMatchSnapshot();
});
