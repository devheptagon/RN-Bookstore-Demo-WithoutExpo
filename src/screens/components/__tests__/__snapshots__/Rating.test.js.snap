// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders correctly 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 5,
      "marginTop": 5,
      "width": "30%",
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
